src/plugins/ui/edition.ts

```diff
@@ -442,14 +446,14 @@ export class EditionPlugin extends UIPlugin {
     }
   }
 
-  private cancelEdition() {
+  private cancelEdition(activateEditionSheet = true) {
     if (this.mode === "inactive") {
       return;
     }
     this.mode = "inactive";
     this.selection.release(this);
     const sheetId = this.getters.getActiveSheetId();
-    if (sheetId !== this.sheetId) {
+    if (activateEditionSheet && sheetId !== this.sheetId) {
```

Comment:
I think it calls for a change of structure. Adding this parameter just to avoid an infinite loop is not that sturdy

Reply:
Created function `cancelEdition()` and `cancelEditionAndActivateSheet()`